dScripts/CppScripts.cpp

```diff
@@ -860,6 +861,36 @@ CppScripts::Script* CppScripts::GetScript(Entity* parent, const std::string& scr
 	else if (scriptName == "scripts\\zone\\LUPs\\WBL_generic_zone.lua")
 		script = new WblGenericZone();
 
+	// pickups
+	if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_1_BRONZE-COIN-SPAWNER.lua")
+		script = new SpecialCoinSpawner(1);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_1_GOLD-COIN-SPAWNER.lua")
+		script = new SpecialCoinSpawner(10000);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_1_SILVER-COIN-SPAWNER.lua")
+		script = new SpecialCoinSpawner(100);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_10_BRONZE-COIN-SPAWNER.lua")
+		script = new SpecialCoinSpawner(10);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_10_GOLD-COIN-SPAWNER.lua")
+		script = new SpecialCoinSpawner(100000);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_10_SILVER-COIN-SPAWNER.lua")
+		script = new SpecialCoinSpawner(1000);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_25_BRONZE-COIN-SPAWNER.lua")
+		script = new SpecialCoinSpawner(25);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_25_GOLD-COIN-SPAWNER.lua")
+		script = new SpecialCoinSpawner(250000);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_25_SILVER-COIN-SPAWNER.lua")
+		script = new SpecialCoinSpawner(2500);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_IMAGINE-POWERUP-SPAWNER.lua")
+		script = new SpecialPowerupSpawner(13);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_IMAGINE-POWERUP-SPAWNER-2PT.lua")
+		script = new SpecialPowerupSpawner(129);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_LIFE-POWERUP-SPAWNER.lua")
+		script = new SpecialPowerupSpawner(5);
+	else if (scriptName == "scripts\\ai\\SPEC\\L_SPECIAL_ARMOR-POWERUP-SPAWNER.lua")
```

Comment:
I dont understand why this doesn't work.  All it does is target caster and repair their armor.  Are you sure that issue is skill related?

Reply:
Yes, see previous commit where I had a log to verify that the skill was being cast. And the effects for the speed boost pay as well, just don't get the boost

Reply:
Hm I'm still quite skeptical of that claim considering how simple the skill is.  Can you verify that the repair behavior is just not repairing or that it's repairing the wrong entity?

Reply:
speed needed to be split out and use target as caster since the pickup itself dies and stops the skill
armor was using the wrong skill, they updated the object but not the script at some point